### PR TITLE
Add debug for hash_map::{Entry, VacantEntry, OccupiedEntry}

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @alexcrichton

### CODE REVIEW

File: src/libstd/collections/hash/map.rs [hunk 1]
```diff
@@ -1351,13 +1351,37 @@ pub enum Entry<'a, K: 'a, V: 'a> {
     ),
 }
 
+#[stable(feature= "debug_hash_map", since = "1.12.0")]
+impl<'a, K: 'a + Debug + Ord, V: 'a + Debug> Debug for Entry<'a, K, V> {
```
Comment (apasel422) [inline src/libstd/collections/hash/map.rs]: These bounds shouldn't mention `Ord`.

**Comment by GuillaumeGomez** [comment]
Updated.

**Pull request merged**